### PR TITLE
add upgrade guide from v3.x to v4.x

### PR DESCRIPTION
**Pull request by kmossco**
This adds an upgrade guide like we have in the other libraries for updating from v3.x to v4.x.

### CODE REVIEW

File: 4.0-Upgrade.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# This will return an array of plans (if available), and a cursor + has_more fields
+{
+    "plans": [
+        {
+            "uuid": "some_uuid",
+            "data_source_uuid": "some_uuid",
+            "name": "Master Plan"
+        }
+    ],
+    "has_more": True,
```
Comment (tudorpavel) [inline 4.0-Upgrade.md:22]: If this is an example JSON file of what the API returns WDYT about moving it to a different code block with `json` formatting?

Also, nitpick:
```suggestion
    "has_more": true,
```

Comment (kmossco) [reply 4.0-Upgrade.md:22]: Nicely spotted, I copied this from Python, didn't notice it. We can move it, I was just copying it from what the other libraries were doing, but it's a good idea.

File: 4.0-Upgrade.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ],
+    "has_more": true,
+    "cursor": "MjAyMy0wNy0yOFQwODowOToyMi4xNTQyMDMwMDBaJjk0NDQ0Mg=="
+}
+```
+
+```ruby
+# You can get the following pages by passing a cursor to .all
+if plans.has_more
+    more_plans = ChartMogul::Plan.all(per_page: 12, cursor: plans.cursor)
```
Comment (tudorpavel) [inline 4.0-Upgrade.md:33]: Sorry, one final nitpick on consistent indentation:
```suggestion
  more_plans = ChartMogul::Plan.all(per_page: 12, cursor: plans.cursor)
```

**Review by tudorpavel — APPROVED** [review]

File: 4.0-Upgrade.md (hunk 2 above)
Comment (kmossco) [reply 4.0-Upgrade.md]: 🦅 👁️

**Pull request merged**